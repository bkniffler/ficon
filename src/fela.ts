//@ts-ignore
import { createComponent } from 'react-fela';

export interface StyledPropTypes {
  color?: string | boolean;
  size: number | string;
  theme?: any;
  onClick?: Function;
  margin?: number | string;
  marginLeft?: number | string;
  marginRight?: number | string;
  marginTop?: number | string;
  marginBottom?: number | string;
}

const getStyle = (props: any) => {
  const {
    theme,
    color,
    size = '1em',
    margin,
    marginLeft,
    marginRight,
    marginTop,
    marginBottom
  } = props;
  return {
    margin,
    marginLeft,
    marginRight,
    marginTop,
    marginBottom,
    width: size,
    height: size,
    fill:
      color === true
        ? theme.color
        : typeof color === 'string'
          ? theme[color] || color
          : theme.inverted
            ? theme.light
            : theme.dark
  };
};

const Svg = createComponent(getStyle, 'svg', [
  'onClick',
  'onMouseEnter',
  'onMouseLeave',
  'onMouseOver',
  'onMouseDown',
  'viewBox',
  'xmlns'
]);

export default Svg;
