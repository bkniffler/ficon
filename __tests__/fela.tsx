import 'jest';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { createRenderer } from 'fela';
import { Provider, ThemeProvider } from 'react-fela';
import { renderToMarkup } from 'fela-dom';

import Svg from '../src/fela';

describe('SVG', () => {
  it('should render correctly (default color)', () => {
    const fela = createRenderer();

    const component = renderer.create(
      <Provider rehydrate renderer={fela}>
        <Svg size={12} />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(renderToMarkup(fela)).toMatchSnapshot('css');
  });

  it('should render correctly (inverted)', () => {
    const fela = createRenderer();

    const component = renderer.create(
      <Provider rehydrate renderer={fela}>
        <ThemeProvider theme={{ inverted: true }}>
          <Svg size={12} />
        </ThemeProvider>
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(renderToMarkup(fela)).toMatchSnapshot('css');
  });

  it('should render correctly (theme color)', () => {
    const fela = createRenderer();

    const component = renderer.create(
      <Provider rehydrate renderer={fela}>
        <ThemeProvider theme={{ color: 'blue' }}>
          <Svg size={12} color />
        </ThemeProvider>
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(renderToMarkup(fela)).toMatchSnapshot('css');
  });

  it('should render correctly (red)', () => {
    const fela = createRenderer();

    const component = renderer.create(
      <Provider rehydrate renderer={fela}>
        <Svg size={12} color="red" />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
    expect(renderToMarkup(fela)).toMatchSnapshot('css');
  });
});
