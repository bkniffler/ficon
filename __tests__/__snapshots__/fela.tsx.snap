// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SVG should render correctly (default color) 1`] = `
<svg
  className="a b"
/>
`;

exports[`SVG should render correctly (default color): css 1`] = `"<style type=\\"text/css\\" data-fela-rehydration=\\"2\\" data-fela-type=\\"RULE\\">.a{width:12}.b{height:12}</style>"`;

exports[`SVG should render correctly (inverted) 1`] = `
<svg
  className="a b"
/>
`;

exports[`SVG should render correctly (inverted): css 1`] = `"<style type=\\"text/css\\" data-fela-rehydration=\\"2\\" data-fela-type=\\"RULE\\">.a{width:12}.b{height:12}</style>"`;

exports[`SVG should render correctly (red) 1`] = `
<svg
  className="a b c"
/>
`;

exports[`SVG should render correctly (red): css 1`] = `"<style type=\\"text/css\\" data-fela-rehydration=\\"3\\" data-fela-type=\\"RULE\\">.a{width:12}.b{height:12}.c{fill:red}</style>"`;

exports[`SVG should render correctly (theme color) 1`] = `
<svg
  className="a b c"
/>
`;

exports[`SVG should render correctly (theme color): css 1`] = `"<style type=\\"text/css\\" data-fela-rehydration=\\"3\\" data-fela-type=\\"RULE\\">.a{width:12}.b{height:12}.c{fill:blue}</style>"`;
